### PR TITLE
impyla connect hs2 with hive auth

### LINKED ISSUE #641: Pass the auth mechanism to impyla connection
**Is your feature request related to a problem? Please describe.**

Since gohive supports to the auth argument: https://github.com/sql-machine-learning/gohive/issues/49, we also need to pass the auth argument to the impyla connection.

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #641

### CODE REVIEW

**Comment by typhoonzero** [comment]
some tests need be fixed

File: sql/codegen.go [hunk 1]
```diff
@@ -305,7 +317,7 @@ database="{{.Database}}"
 database=""
 {{end}}
 
-conn = connect(driver, database, user="{{.User}}", password="{{.Password}}", host="{{.Host}}", port={{.Port}})
+conn = connect(driver, database, user="{{.User}}", password="{{.Password}}", host="{{.Host}}", port={{.Port}}, auth={{.Auth}})
```
Comment (tonyyang-svail) [inline sql/codegen.go]: `auth={{.Auth}}` => `auth="{{.Auth}}"`

File: cmd/sqlflowserver/main_test.go [hunk 2]
```diff
@@ -268,7 +268,7 @@ func TestEnd2EndHive(t *testing.T) {
 	if testDBDriver != "hive" {
 		t.Skip("Skipping hive tests")
 	}
-	dbConnStr = "hive://127.0.0.1:10000/iris"
+	dbConnStr = "hive://127.0.0.1:10000/iris?auth=NOSASL"
```
Comment (typhoonzero) [inline cmd/sqlflowserver/main_test.go:271]: Maybe add more tests to test connection string with out `?auth=NOSASL`?

**Review by typhoonzero — APPROVED** [review]

File: cmd/sqlflowserver/main_test.go (hunk 2 above)
Comment (Yancey0623) [reply cmd/sqlflowserver/main_test.go:271]: Will add it in the next pr.

File: sql/codegen.go (hunk 1 above)
Comment (Yancey0623) [reply sql/codegen.go]: done.

**Pull request merged**